subprojects/base-services/src/main/java/org/gradle/internal/hash/Hashing.java

```diff
@@ -231,29 +226,16 @@ public void putBytes(byte[] bytes, int off, int len) {
             digest.update(bytes, off, len);
         }
 
-        @Override
-        public HashCode hash() {
-            done = true;
-            byte[] bytes = digest.digest();
-            return HashCode.fromBytesNoCopy(bytes);
-        }
-
-        private void update(int length) {
-            checkNotDone();
-            digest.update(buffer.array(), 0, length);
-            BufferCaster.cast(buffer).clear();
-        }
-
         @Override
         public void putInt(int value) {
-            buffer.putInt(value);
-            update(4);
+            byte[] bytes = Ints.toByteArray(value);
```

Comment:
changed from little-endian to big-endian, for consistency

Reply:
Looks like changing endianness breaks cross-version compatibility: https://builds.gradle.org/viewLog.html?buildId=22467137&buildTypeId=Gradle_Check_Stage_QuickFeedbackLinuxOnly_Trigger